mcstatus/querier.py

```diff
@@ -131,7 +134,7 @@ def __init__(self, raw, players):
             raise ValueError("The provided data is not valid")
 
     @classmethod
-    def from_connection(cls, response: Connection):
+    def from_connection(cls, response: Connection) -> Self:
```

Comment:
Importing annotations from future should allow:
```suggestion
    def from_connection(cls, response: Connection) -> ServerQuerier:
```

This removes the need for `Self`, which would also remove `TYPE_CHECKING`.

Reply:
Yes, hiwever the whole point or `Self` is that it's extensible. Using the class name isn't good when another class would inherit from this one, since at that point the return type should be that new class, not the old one.
```py
class Foo:
    @classmethod
    def test_1(cls) -> Self:
        ...

    @classmethod
    def test_2(cls) -> Foo:
        ...

class Bar(Foo):

    def only_in_bar():
        ...

reveal_type(Foo.test_1())  # Foo
reveal_type(Foo.test_2())  # Foo
reveal_type(Bar.test_1())  # Bar (Self changed to Bar)
reveal_type(Bar.test_2())  # Foo! But it should be Bar
```
This problem is the whole reason `Self` was inplemented. While returning as the superclass isn't necessarily wrong type-wise, as we can type-hint an object with it's supertype instead of the actual type, however when doing that, if this true type has additional functionality that's not in that supertype, the type checker would complain if we tried to access it, even though it should absoletely be there, since we didn't initialize with `Foo()`, but with `cls()`, which changes from class to class.

Reply:
Today I learned